File: Algorithms/Sort/selection_sort/selection_sort.cpp
#include "../sort.h"


void SelectionSort(int* array, uint length, bool isDescend)
{
    int selection;
    int temp;
    for (int i = 0; i != length; i++) {
        selection = i;
        for (int j = i + 1; j != length; j++) {
            if ((array[selection] > array[j]) ^ isDescend) {
                selection = j;
            }
        }
        temp = array[selection];
        array[selection] = array[i];
        array[i] = temp;
    }
}

File: algorithms/sort/selection_sort/selection_sort.py
from typing import List


def selection_sort(array: List[int], is_descend: bool = False):
    for i in range(len(array)):
        selection = i
        for j in range(i+1, len(array)):
            if (array[selection] > array[j]) ^ is_descend:
                selection = j
        array[selection], array[i] = array[i], array[selection]
